src/basic/pidref.h

```diff
@@ -1,10 +1,7 @@
 /* SPDX-License-Identifier: LGPL-2.1-or-later */
 #pragma once
 
-typedef struct PidRef PidRef;
-
 #include "macro.h"
-#include "process-util.h"
```

Comment:
Hmm, what about PIDREF_AUTOMATIC here?

I'm not entirely sure what this is supposed to achieve? Conversely this one no longer includes everything it needs with this change?

Reply:
I moved `PID_AUTOMATIC` to constants.h for now to fix that problem.

The goal is to get rid of the circular includes between the two. Like I mentioned in the description, they're just really confusing for both users and tooling so I think we should get rid of them everywhere.

Reply:
I think that's completely fine to move it. It is very much a constant.